components/config-model-migrator/org.wso2.is.configuration.diff.creator/src/main/java/org/wso2/is/configuration/diff/creater/ConfigurationDiffChecker.java

```diff
@@ -0,0 +1,356 @@
+                    .withNodeMatcher(new DefaultNodeMatcher(ElementSelectors.byName))
+                    .build();
+        } catch (XMLUnitException e) {
+            log.error("Error occurred when comparing default File : " + defaultFile.getName() + " and migrated" +
+                    " file :" + migratedFile.getName());
+        }
+        return null;
+    }
+
+    /**
+     * Compare difference to print the output.
+     *
+     * @param migratedFile    Migrated xml file.
+     * @param detailedDiff    Diff of all the difference.
+     * @param existingXMLTags Map of existing xml tags.
+     * @param keyValues       Map of key values.
+     * @param outputGenerator OutputGenerator.
+     */
+    private void getDifferenceToMaps(File migratedFile,
+                                     Diff detailedDiff,
+                                     Map<String, String> existingXMLTags,
+                                     Map<String, String> keyValues,
+                                     OutputGenerator outputGenerator) {
+
+        if (detailedDiff != null) {
+            Iterable<Difference> differences = detailedDiff.getDifferences();
+            for (Difference difference : differences) {
+                if (difference.getResult().name().equals("DIFFERENT")) {
+                    String csvEntry = null;
+                    String csvKey;
+                    String defaultValue = " ";
+                    String changedValue = " ";
+
+                    csvKey = difference.getComparison().getControlDetails().getXPath();
+                    if (!isMigratedXPathInDiffContainsText(difference)) {
+                        if (isMigratedFileNotContainPropertiesWhichDefaultHas(difference)) {
+                            continue;
+                        } else {
+                            csvEntry =
+                                    migratedFile.getName().concat(MigrationConstants.CSV_SEPARATOR_APPENDER)
+                                            .concat(MigrationConstants.XML_FILE_TYPE)
+                                            .concat(MigrationConstants.CSV_SEPARATOR_APPENDER).concat(csvKey)
+                                            .concat("| | | |").concat(defaultValue).concat("| | Not In Default");
+                        }
+                    }
+
+                    if (StringUtils.isNotBlank(csvKey)) {
+                        if (difference.getComparison().getControlDetails().getTarget() != null) {
+                            changedValue = difference.getComparison().getControlDetails().getTarget().getNodeValue();
+                        }
+                        if (isContainDefaultValue(difference)) {
+                            defaultValue = difference.getComparison().getTestDetails().getTarget().getNodeValue();
+                        }
+                        if (isXpathNotInExistingTags(existingXMLTags, csvKey)) {
+                            if (StringUtils.isBlank(csvEntry)) {
```

Comment:
we can improve here as` if (isXpathNotInExistingTags(existingXMLTags, csvKey) && StringUtils.isBlank(csvEntry))`

Reply:
we can not join this because we have some task to do if this is not Blank, if this csvEntry is blank we care adding new one